### PR TITLE
sign user up

### PR DESCRIPTION
**Pull request by zaher-aa**
## Solves #15
- add signup queries
- created signup controller
- added test suites for signup with different inputs
- edited email in the database to be unique
- edited password for fake data to be a hashed one

### CODE REVIEW

**Review by khaled-samy — DISMISSED** [review]

File: server/controllers/auth/signup.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { signupSchema, signToken } from '../../utils';
+import { CustomError } from '../../errors';
+
+export default async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const {
+      role, email, username, password, bio, avatar,
+    } = req.body;
+
+    await signupSchema.validate({ ...req.body, role });
```
Comment (khaled-samy) [inline server/controllers/auth/signup.ts:13]: Why you send role alone from the body, even though it's part of it?

**Review by khaled-samy — COMMENTED** [review]
please check my reviews

File: server/controllers/auth/signup.ts (hunk 1 above)
Comment (zaher-aa) [reply server/controllers/auth/signup.ts:13]: Oooh, you're right 👍

**Review by amjed-98 — APPROVED** [review]

File: __test__/index.test.ts [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+
+beforeEach(dbBuild);
+afterAll(() => dbConnection.end());
+
+describe('GET /', () => {
+  it('should return 200 OK and Content-Type /json/', async () => {
+    const res = await supertest(app).get('/');
+    expect(res.status).toBe(200);
+    expect(res.type).toBe('application/json');
+    expect(res.body.message).toBe('Server Is Running');
```
Comment (Mu7ammadAbed) [inline __test__/index.test.ts:14]:
```suggestion
    const res = await supertest(app)
    .get('/')
    .expect('Content-type', /json/)
    .expect(200);
    expect(res.body.message).toBe('Server Is Running');
```

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by amjed-98 — APPROVED** [review]

**Review by khaled-samy — APPROVED** [review]

**Pull request merged**